### ISSUE TITLE
Preview to emails from contributions list does not respect roles

### ISSUE THREAD

**Issue by JohannesBuchner** [indico/indico#6520]
**Describe the bug**
Previewing an email shows wrong addresses.

**To Reproduce**
1. Go to Contribution list, select contributions
2. Click "email"
3. select role "Speaker"
4. Click "Preview"
5. The "Dear {first_name}" may be showing a co-author, which is wrong
6. Click Submit -> the previewed email is not actually sent

**Expected behavior**
Preview should show an example email that will actually be sent (in the example above, to a speaker). The role selection should be respected.

**Comment by tomasr8** [comment]
I don't think this is a bug, you can have co-authors who are also speakers and in that case the email is sent to them as well.

**Comment by JohannesBuchner** [comment]
But it wasn't.

**Comment by tomasr8** [comment]
Yes you are right. After checking the code, it's because we don't use the role when creating the preview, we simply use the first person regardless of the role selected. I think it makes sense to improve that

**Comment by JohannesBuchner** [comment]
Please also see https://github.com/indico/indico/issues/6460
